src/core/views.py

```diff
@@ -1640,53 +1640,50 @@ def serve_all_review_files(request, submission_id, review_type):
 
     zip_subdir = "Files of Submission #%s" % submission_id
     zip_filename = "%s.zip" % zip_subdir
-    s = StringIO.StringIO()  # Open StringIO to grab in-memory ZIP contents.
-    zf = zipfile.ZipFile(s, "w")  # The zip compressor.
+    zip_buffer = io.BytesIO()  # Open BytesIO to grab in-memory ZIP contents.
 
-    if review_type == "external":
-        for _file in external_review_files:
-            fpath = os.path.join(
-                settings.BOOK_DIR,
-                submission_id,
-                _file.uuid_filename,
-            )
-            fdir, fname = os.path.split(fpath)
-            zip_path = os.path.join(zip_subdir, fname)
-            zf.write(fpath, zip_path)
+    with zipfile.ZipFile(zip_buffer, "w") as zip_archiver:
 
-    else:
-        for _file in internal_review_files:
-            fpath = os.path.join(
-                settings.BOOK_DIR,
-                submission_id,
-                _file.uuid_filename,
-            )
-            fdir, fname = os.path.split(fpath)
-            zip_path = os.path.join(zip_subdir, fname)
-            zf.write(fpath, zip_path)
-
-    zf.close()
-    fd = default_storage.open(os.path.join(
-        settings.BOOK_DIR, submission_id, zip_filename),
-        'wb'
-    )
-    fd.write(s.getvalue())
-    fd.close()
-    fsock = default_storage.open(
-        os.path.join(
-            settings.BOOK_DIR,
-            submission_id,
-            zip_filename
-        ),
-        'r'
-    )
-    resp = StreamingHttpResponse(
-        fsock,
-        content_type="application/x-zip-compressed"
+        if review_type == "external":
```

Comment:
Very minor, but is the only difference between the `if` and `else` that you either loop through `external_review_files` or `internal_review_files`?

Reply:
That's very weird. I wouldn't have expected that even of the Rua codebase.